packages/core/src/payment/payment-method-request-sender.ts

```diff
@@ -44,4 +50,79 @@ export default class PaymentMethodRequestSender {
             params,
         });
     }
+
+    /**
+     * GraphQL payment requests
+     */
+    loadPaymentWalletWithInitializationData(
+        methodId: string,
+        options: HeadlessPaymentRequestOptions,
+    ): Promise<Response<PaymentMethod>> {
+        const entityId = this.getPaymentEntityId(methodId);
+
+        const graphQLQuery = `
+            query {
+                site {
+                    paymentWalletWithInitializationData(filter: {paymentWalletEntityId: "${entityId}"}) {
+                        clientToken
+                        initializationData
+                    }
+                }
+            }
+        `;
+
+        const requestOptions: HeadlessPaymentRequestOptions = {
+            headers: {
+                ...options.headers,
+                'Content-Type': 'application/json',
+            },
+            body: {
+                query: graphQLQuery,
+            },
+        };
+
+        return this._requestSender
+            .post<HeadlessPaymentMethodResponse>('/graphql', requestOptions)
+            .then((response) => this.transformToPaymentMethodResponse(response, methodId));
+    }
+
+    private transformToPaymentMethodResponse(
```

Comment:
```suggestion
    private _transformToPaymentMethodResponse(
```

Reply:
Also maybe refactor this to a separate transformer file.

Reply:
Moved to the separate file